### ISSUE TITLE
duplicate To headers from latest version

### ISSUE THREAD

**Issue by jimfonseca** [pivotal-cf/email-resource#47]
Hi, we've been using pcfseceng/email-resource on our team for a couple years and it's been great. For each pipeline, we have been emailing one or more developers. Around mid-August, we started receiving emails from sendgrid support on all pipelines which have multiple recipients in the "to_text" field.

The emails from sendgrid have the subject line "Invalid SMTPAPI Header"
and body:

> A message was received from this address by our systems that had errors in the SMTPAPI header, and cannot be processed due to the following:
> 
>   - 2 duplicate to header detected, rfc5322 states that there should be a max of 1
> 
> If you require assistance in working with the SendGrid API, please contact us at support@sendgrid.com

Here is an excerpt from an affected pipeline with some simplifications/generalizations

```
resource_types:
  - name: email
    type: docker-image
    source:
      repository: pcfseceng/email-resource
...

  - name: send-an-email
    type: email
    source:
      smtp:
        host: ((smtp-host))
        port: ((smtp-port))
        username: apikey
        password: ((sendgrid-api-key))
      from: "ourteam@foo.com"
...
    on_success:
      in_parallel:
        - put: send-an-email
          params:
            subject_text: "our subject"
            body_text: "body text"
            to_text: [ "alice@foo.com", "bob@foo.com" ]
```

Concourse indicates that it is indeed using the image with the "latest" tag
`fb79dc8e3d7d03379d1592faa16ea46e6c7454e63688c94e28962262b45a9205`
which matches the sha here: [https://hub.docker.com/layers/pcfseceng/email-resource/latest/images/sha256-fb79dc8e3d7d03379d1592faa16ea46e6c7454e63688c94e28962262b45a9205?context=explore](https://hub.docker.com/layers/pcfseceng/email-resource/latest/images/sha256-fb79dc8e3d7d03379d1592faa16ea46e6c7454e63688c94e28962262b45a9205?context=explore)

The emails from sendgrid support have an attachment showing the email received from pcfseceng/email-resource, which seems to indicate the multiple "To" headers.

> Received: by filterdrecv-7bf5c69d5-kwxgp with SMTP id filterdrecv-7bf5c69d5-kwxgp-1-614A67CC-43
>         2021-09-21 23:16:28.567144502 +0000 UTC m=+1731390.697352871
> Received: from localhost (unknown [35.193.20.185])
> 	by geopod-ismtpd-4-4 (SG) with ESMTP id N7aNpXgfRc2yJ4OPN8HUcA
> 	Tue, 21 Sep 2021 23:16:28.308 +0000 (UTC)
> From: ourteam@foo.com
> Date: Tue, 21 Sep 2021 23:16:27 +0000
> Subject: our subject
> To: alice@foo.com
> To: bob@foo.com
> Content-Type: multipart/mixed;
> 	boundary="0b6b34c35065cb31646bbd4d74b6d9d6ab998c6efbeb8816e65f37b9dc27"; charset=UTF-8
> MIME-Version: 1.0
> 
> --0b6b34c35065cb31646bbd4d74b6d9d6ab998c6efbeb8816e65f37b9dc27
> Content-Type: multipart/alternative;
> 	boundary="4beee005b04ff286d9a68e1efc9f08c75136ec7268221d0265c05de9ab4c"
> 
> --4beee005b04ff286d9a68e1efc9f08c75136ec7268221d0265c05de9ab4c
> Content-Transfer-Encoding: quoted-printable
> Content-Type: text/plain; charset=UTF-8
> 
> body text
> 
> --4beee005b04ff286d9a68e1efc9f08c75136ec7268221d0265c05de9ab4c--
> 
> --0b6b34c35065cb31646bbd4d74b6d9d6ab998c6efbeb8816e65f37b9dc27--

We followed up with SendGrid and they replied with the following:

> Hello
> 
> Thank you for contacting Twilio SendGrid, this is Sachin and I'll be glad to help you today.
> 
> As we continue to modernize and monitor our email delivery infrastructure to comply with industry standards, we are changing a few behaviours on the API. We had notificed about this change in June/July via email sent to the email address we have on file. The changes are as follows:
> 
> 1.    The  word “enabled” (instead of “enable”) will no longer be a recognized parameter for the Mail Send API. The “enable” parameter is required, and if the parameter is omitted, settings will not be applied. Please use “enable” to ensure the desired behaviour.
> 
> 2.    Only one of each type of header is permitted.  We will no longer support duplicate headers for the following:
> 
> ·         from
> 
> ·         sender
> 
> ·         reply-to
> 
> ·         to
> 
> ·         cc
> 
> ·         bcc
> 
> ·         x-smtpapi
> 
> 3.    We will not apply any personalization block substitutions to your custom argument fields in the V3 endpoint. 
> 
> 
> What do you need to do?
> 
> To avoid disruption to your mail send service, please be sure to take the following actions before August 10, 2021: 
> 
> 1.    Check your code that uses the mail send endpoint and ensure that the “enable” parameter is included under the filter when applicable.  
> 
> 2.    Check your mail send header to ensure that you are using just one X-SMTPAPI header and address header of the same kind. Remove multiple headers of the same kind, so you have only 1 header of the same kind. For example, if you are currently using multiple “from” headers, you should modify your code so that you have a single “from” header. 
> 
> 3.    Check your code to ensure that you are not applying any personalization block substitution to your custom argument fields.

We'd greatly appreciate any guidance or help to resolve this. 
Thanks,
Jim

**Comment by calebwashburn** [comment]
@jimfonseca Pushed a new dev build of this if you can validate that this is working as expected will merge and release 1.0.20 version of this.  To update your concourse pipeline just change the image tag to ```dev-build```


```
resource_types:
  - name: email
    type: docker-image
    source:
      repository: pcfseceng/email-resource
      tag: dev-build
```

Let me know if you have any problems as looking to do more cleanup and address the other issues reported.

**Comment by jimfonseca** [comment]
Many thanks Caleb. I'm unable to test this but I will forward this on to my former colleagues.

**Comment by lyle-ascn** [comment]
We have been testing the new dev-build for the last week and it looks like it fixed our issue with multi email address for us.

**Comment by calebwashburn** [comment]
Fixed in latest docker image which aligns to v1.0.20

**Comment by jrobison-sb** [comment]
@calebwashburn we're seeing a similar rejection (from Gmail), even after bumping our resource image to 1.0.20. Here is the rejection log from Gmail:

```
Received from an SMTP server with IP address: 1.2.3.4 (TLS enabled)
550-5.7.1 [1.2.3.4] This message is not RFC 5322 compliant, the issue is: 550-5.7.1 duplicate To headers. To reduce the amount of spam sent to Gmail, 550-5.7.1 this message has been blocked. Please review 550 5.7.1 RFC 5322 specifications for more information.
```

Here are the headers shown with `debug: "true"`, where you can see that the `To` header is indeed duplicated:

```
2022/11/08 14:50:19 Writing message to SMTP Server From: devops@mycompany.com
Mime-Version: 1.0
Date: Tue, 08 Nov 2022 14:50:19 +0000
Subject: AWS daily spend by service
To: devops@mycompany.com
To: aperson@mycompany.com
```

And here is how we're using the resource in a pipeline:

```
---
resource_types:

- name: email
  type: docker-image
  source:
    repository: pcfseceng/email-resource
    tag: 1.0.20
    username: ((docker_hub_username))
    password: ((docker_hub_access_token))

resources:

  - name: send-an-email
    type: email
    source:
      smtp:
        host: email-smtp.us-east-1.amazonaws.com
        port: "587"
        username: ((concourse_ses_smtp_username))
        password: ((concourse_ses_smtp_password))
      from: devops@mycompany.com
      to:
        - devops@mycompany.com
        - aperson@mycompany.com

jobs:
  - name: daily-spend-by-service
    plan:
    ...
      - put: send-an-email
        params:
          subject_text: AWS daily spend by service
          body_text: See attached for AWS daily spend by service report.
          debug: "true"
          attachment_globs:
            - devops-repo/concourse/cost-explorer/email-out/graph.png
            - devops-repo/concourse/cost-explorer/email-out/aws_daily_spend_by_top_services.csv
```

FWIW I also tried reworking the pipeline to get rid of the `to` array in the resource, and replace it with a `to_text` field in the `put` step (`to_text: devops@mycompany.com,aperson@mycompany.com`), but Gmail still rejected it with the same message.

We were only able to work around this by moving one of the address into a `cc` key.

**Comment by calebwashburn** [comment]
@jrobison-sb released just now 1.0.21/latest and this is fixed.  Something happened in the merge into main and this didn't get added.   Let me know if still an issue.

**Comment by jrobison-sb** [comment]
1.0.21 fixed the issue that we saw. Thank you.
